scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -519,10 +519,33 @@ def test_random_generator(self):
             differential_evolution(self.quadratic,
                                    [(-100, 100)],
                                    polish=False,
-                                   seed=rng,
+                                   rng=rng,
                                    tol=0.5,
                                    init=init)
 
+    def test_rng_seed_spec007(self):
```

Comment:
Should we test every function we apply this to or have a central test on a dummy function?

Reply:
We should have a good central test that exercises the decorator. This does to a certain extent. I don't think we necessarily need to do it for every function. However, there may be remnant code that wasn't written for `Generator`, in which case some extra testing wouldn't go amiss.